demos/python_demos/mri_segmentation.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python
+"""
+ Copyright (c) 2019 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+"""
+OpenVINO Python Inference Script
+This will load the OpenVINO version of the model (IR)
+and perform inference on a few validation samples
+from the Medical Decathlon dataset. The Medical
+Decathlon dataset (http://medicaldecathlon.com/) is
+distributed under the Creative Commons Attribution-ShareAlike 
+International 4.0 license
+(https://creativecommons.org/licenses/by-sa/4.0/).
+
+You'll need the extension library to handle the Resize_Bilinear operations.
+
+python inference_openvino.py -l ${INTEL_CVSDK_DIR}/inference_engine/lib/centos_7.4/intel64/libcpu_extension_avx2.so
+
+"""
+
+import sys
+import os
+from argparse import ArgumentParser
+import numpy as np
+import logging as log
+from time import time
+from openvino.inference_engine import IENetwork, IEPlugin
+
+def dice_score(pred, truth):
```

Comment:
The additive smooth is essential for protecting against the corner case of an empty mask and prediction (which should be Dice = 1). Refer to the paper https://arxiv.org/pdf/1707.03237.pdf or the section on additive/Laplace smoothing in Wikipedia (https://en.wikipedia.org/wiki/Additive_smoothing). For popular implementations, please refer to https://github.com/ellisdg/3DUnetCNN/blob/master/unet3d/metrics.py  or https://github.com/jakeret/tf_unet/blob/bd2c940f2f73c1b57ca2397d6d200c72f0e4e4bc/tf_unet/unet.py

Reply:
Additive smoothing is used in order to make this function differentiable. In return, this so-called Dice loss function (all of the above-stated references) is different from Dice index in cases with small gt masks (with the size of few pixels), and it adds bias to the final evaluation. As you stated, empty masks should produce Dice index equals to 1 indeed. However, we are not constrained by the differentiability requirement, thus we can handle empty masks separately with if-else statement.

Reply:
That's a fair statement, but I think it is only relevant when we are trying to show that our model is somehow state of the art. Why is this an issue in the model zoo? Aren't we just showing examples of how to use OV? I'm just as much of an expert on this as you and we simply disagree here.

Reply:
Let me try to mediate =)

I guess, technically, the name of the function is misleading. If it were `dice_loss` then it would be, technically, correct. Now, I think, there's 2 options - either we rename the function or introduce `if'`s (as @lachinov suggests). In terms of implementation both are equally simple. 

Personally, I think, I'm leaning towards additional `if`s, but in the end, it's your call, @tonyreina . Either way should work.